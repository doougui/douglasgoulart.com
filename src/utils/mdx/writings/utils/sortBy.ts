import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
