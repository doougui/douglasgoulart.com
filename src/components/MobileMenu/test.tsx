import { render, screen } from 'utils/tests';

import { MobileMenu } from '.';

jest.mock('./components/Navigation', () => ({
  __esModules: true,
  Navigation: () => <div data-testid="navigation" />,
}));

jest.mock('./components/Toggle', () => ({
  __esModules: true,
  Toggle: () => <div data-testid="toggle" />,
}));

jest.mock('framer-motion', () => ({
  ...jest.requireActual('framer-motion'),
  useReducedMotion: () => true,
}));

describe('<MobileMenu />', () => {
  it('should render only on mobile devices', () => {
    render(<MobileMenu />);

    expect(screen.getByTestId('navigation').parentElement).toHaveStyleRule(
      'display',
      'none',
      {
        media: '(min-width: 768px)',
      },
    );

    expect(screen.getByTestId('navigation').parentElement).not.toHaveStyleRule(
      'display',
      'none',
      {
        media: '(max-width: 768px)',
      },
    );
  });

  it('should render navigation and toggle button', () => {
    render(<MobileMenu />);

    expect(screen.getByTestId('navigation')).toBeInTheDocument();
    expect(screen.getByTestId('toggle')).toBeInTheDocument();
  });
});
