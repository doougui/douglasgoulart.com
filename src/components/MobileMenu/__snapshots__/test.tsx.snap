// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileMenu /> should render navigation and toggle button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
  data-testid="container"
>
  <div
    data-testid="navigation"
  />
  <div
    data-testid="toggle"
  />
</div>
`;
